libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts

```diff
@@ -0,0 +1,51 @@
+import 'reflect-metadata';
+import { BaseValue } from '../../persistence/domain/base-classes/base-value';
+import { Type } from '../../../domain/public_api';
+
+export function instanceForType<T>(Type: Type<T>, fromState = {}) {
+    const newState = new Type();
+
+    return new Proxy(newState, new StateHandler(fromState));
+}
+/**
+ *  Proxy wrapper
+ */
+class StateHandler<TModel> {
+    constructor(
+        private targetState?: TModel
+    ) {
+    }
+
+    get(target: any, prop: PropertyKey): any {
+        // Check is an array and wraps a new one proxy
+        if (['[object Object]', '[object Array]']
+            .indexOf(Object.prototype.toString.call(this.targetState[prop])) > -1) {
+            return new Proxy(this.targetState[prop], this);
+        }
+        return target[prop] || this.targetState[prop];
+    }
+
+    set<T>(target: TModel, key: PropertyKey, value: Type<T> | Array<Type<T>>): boolean {
```

Comment:
As for me, there is no much sense in `T` in this case.
Let's handle `value` as `unknown` and use `instanceof` to guess what it actually is.